xarray/core/common.py

```diff
@@ -1078,14 +1078,25 @@ def _resample(
         )
 
         grouper: Resampler
-        if isinstance(freq, str):
+        if isinstance(
+            freq,
+            str
+            | datetime.timedelta
+            | pd.Timedelta
+            | pd.offsets.BaseOffset
+            | BaseCFTimeOffset,
```

Comment:
```suggestion
            | pd.DateOffset,
```

Why not `pd.DateOffset`. We haven't yet exposed CFTimeOffset (https://github.com/pydata/xarray/issues/5687) so that will have to wait.

Reply:
I originally thought that BaseOffset was more general but turns out it's not. I've fixed this up now.